ratchet/ratchet.go

```diff
@@ -89,11 +91,17 @@ func EmitAttribute(attrID string, entityID string, entityType string, terraformA
 	return ""
 }
 
-func EmitAttributes(entityID string, entityType string, terraformAttributes gjson.Result) string {
+func EmitDatasources(entityID string, terraformAttributes gjson.Result) string {
 	output := []string{}
 	required := map[string]gjson.Result{}
 	optional := map[string]gjson.Result{}
 	terraformAttributes.ForEach(func(attrID, terraformAttribute gjson.Result) bool {
+		if terraformAttribute.Get("optional").Bool() && terraformAttribute.Get("computed").Bool() {
+			//log.Fatalf("(datasource) %s: Attribute %q is both optional and computed: %v", entityID, attrID, terraformAttribute)
```

Comment:
IIRC optional and computed is a legal combination. I /think/ it means "you can provide this, but it's available for runtime expressions even if you don't".

Reply:
makes sense!